### PR TITLE
Feat/increase data fetching range

### PR DESCRIPTION
**Pull request by IgboPharaoh**
WIP

- [x] Increase data fetching range by 1 year #15 
- [x] Add month markers to grid #57 
- [x] Order organizations by contribution level #48 
- [x] Exclude null contributions
- [x] fix years re-render issue

### CODE REVIEW

File: src/graphql/queries.ts [hunk 1]
```diff
@@ -64,3 +64,87 @@ query ($username: String!){
...
+        createdAt
+        author {
+          login
+        }
+      }
+    }
+  }
+}`
+
+export const FETCH_CONTTRIBUTION_YEARS = `query contributions($username: String!, ) {
```
Comment (Extheoisah) [inline src/graphql/queries.ts:144]: nit -> we already have a query named contributions on line 68. we can call this contributtionYears
```suggestion
export const FETCH_CONTTRIBUTION_YEARS = `query contributionYears($username: String!, ) {
```

File: src/helpers/get-comments.ts [hunk 2]
```diff
@@ -1,17 +1,19 @@
...
     data,
     username
 }: {
-    data: any
+    data: IssueCommentNodes[]
     username: string
 }): Comment[] => {
-    const rawComments = data?.issueComments?.nodes.filter(
-        (comment: any) => comment.issue.author.login === username
+    data = data !== undefined ? data : []
```
Comment (Extheoisah) [inline src/helpers/get-comments.ts:10]:
```suggestion
    data = data ?? []
```

File: src/helpers/get-comments.ts [hunk 3]
```diff
@@ -34,14 +36,16 @@ export const getOthersComments = ({
     data,
     username
 }: {
-    data: any
+    data: IssueCommentNodes[]
     username: string
 }): Comment[] => {
-    const rawComments = data?.issueComments?.nodes.filter(
-        (comment: any) => comment.issue.author.login !== username
+    data = data !== undefined ? data : []
```
Comment (Extheoisah) [inline src/helpers/get-comments.ts:42]:
```suggestion
    data =  data ?? []
```

File: src/helpers/get-comments.ts [hunk 4]
```diff
@@ -60,12 +64,16 @@ export const getOthersComments = ({
...
-export const getLongComments = ({ data }: { data: any }): Comment[] => {
-    const rawComments = data?.issueComments?.nodes.filter(
-        (comment: any) => comment.body.length > 500
-    )
+export const getLongComments = ({
+    data
+}: {
+    data: IssueCommentNodes[]
+}): Comment[] => {
+    data = data !== undefined ? data : []
```
Comment (Extheoisah) [inline src/helpers/get-comments.ts:72]:
```suggestion
    data = data ?? []
```

File: src/helpers/get-comments.ts [hunk 5]
```diff
@@ -60,12 +64,16 @@ export const getOthersComments = ({
...
+
+    const rawComments = data?.filter((comment) => comment.body.length > 500)
```
Comment (Extheoisah) [inline src/helpers/get-comments.ts:74]: no hardcoded value. 500 should be a constant in config
```suggestion
    const rawComments = data?.filter((comment) => comment.body.length > MAX_COMMENT_LENGTH
```

File: src/helpers/get-contribution-years.ts [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+            body: JSON.stringify({
+                query: FETCH_CONTTRIBUTION_YEARS,
+                variables: {
+                    username
+                }
+            })
+        })
+
+        if (!res.ok) {
+            console.log("res", res)
```
Comment (Extheoisah) [inline src/helpers/get-contribution-years.ts:29]:
```suggestion
```

File: src/helpers/get-contribution-years.ts [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+        if (data.errors) {
+            return { error: data.errors, message: "Failed to fetch API" }
+        }
+
+        return {
+            data: contributionYears,
+            error: null
+        }
+    } catch (error) {
+        console.log(error)
```
Comment (Extheoisah) [inline src/helpers/get-contribution-years.ts:48]:
```suggestion
```

File: src/helpers/get-issues-data.ts [hunk 8]
```diff
@@ -0,0 +1,138 @@
...
+        case "FETCH_RANGED_COMMENTS":
+            graphqlQuery = FETCH_RANGED_COMMENTS
+            break
+        default:
+            throw new Error("Invalid query")
+    }
+
+    let hasNextPage = true
+
+    endCursor = startDate?.slice(0, 4) === "2022" ? endCursor : null
```
Comment (Extheoisah) [inline src/helpers/get-issues-data.ts:34]: we should fetch the current year, do a backdate by one year and use that here so we don't have to change years every year

File: src/helpers/get-issues-data.ts [hunk 9]
```diff
@@ -0,0 +1,138 @@
...
+
+    let startCursorObj: any = {}
+    let endCursorObj: any = {}
+
+    let allIssueComments: Array<IssueCommentNodes> = []
+
+    while (hasNextPage) {
+        if (
+            startDate?.slice(0, 4) !== "2023" &&
```
Comment (Extheoisah) [inline src/helpers/get-issues-data.ts:43]: use current year function. no hardcoded value

File: src/helpers/get-issues-data.ts [hunk 10]
```diff
@@ -0,0 +1,138 @@
...
+
+            const data = await res.json()
+            const jsonData: IssueCommentDataType = data.data.user
+
+            const nodes_data = jsonData?.issueComments?.nodes
+            const start_cursor = jsonData?.issueComments?.pageInfo?.startCursor
+            const isNextPage = jsonData?.issueComments?.pageInfo?.hasNextPage
+            const end_cursor = jsonData?.issueComments?.pageInfo?.endCursor
+
+            const prevYear = Number(startDate?.slice(0, 4)) - 1
```
Comment (Extheoisah) [inline src/helpers/get-issues-data.ts:80]: why are slicing just the 1st three values? put a comment here to explain why so its descriptive for anyone looking at the code

File: src/helpers/get-issues-data.ts [hunk 11]
```diff
@@ -0,0 +1,138 @@
...
+
+            const findNextYear = nodes_data?.find(
+                (x) => x.createdAt.slice(0, 4) === prevYear.toString()
```
Comment (Extheoisah) [inline src/helpers/get-issues-data.ts:83]: same here. add a comment explaining the slice at 0, 4

File: src/helpers/get-issues-data.ts [hunk 12]
```diff
@@ -0,0 +1,138 @@
...
+            )
+
+            if (
+                findNextYear ||
+                (isNextPage === false && findNextYear === undefined)
+            ) {
+                const filter_node_data = nodes_data.filter((x) => {
+                    return x.createdAt.slice(0, 4) === startDate?.slice(0, 4)
```
Comment (Extheoisah) [inline src/helpers/get-issues-data.ts:91]: comment here as well

File: src/helpers/get-pull-requests.ts [hunk 13]
```diff
@@ -1,104 +1,112 @@
...
-    data: any
+    data: PrNodes[]
     username: string
 }): PullRequests => {
-    const openPRsData = data.pullRequests.edges.filter(
-        (pr: any) => pr.node.closed === false
-    )
-    const openPRs: PR[] = openPRsData.map((pr: any) => ({
-        totalComments: pr.node.totalCommentsCount,
+    data = data !== undefined ? data : []
```
Comment (Extheoisah) [inline src/helpers/get-pull-requests.ts:12]:
```suggestion
    data = data ?? []
```

File: src/helpers/get-pull-requests.ts [hunk 14]
```diff
@@ -1,104 +1,112 @@
...
+
+    const openPRsData = data?.filter((pr) => pr?.pullRequest?.closed === false)
```
Comment (Extheoisah) [inline src/helpers/get-pull-requests.ts:14]: not sure about these optional chaining. can't we do a check for it the data is null and return. empty array maybe??

File: src/helpers/get-pull-requests.ts [hunk 15]
```diff
@@ -1,104 +1,112 @@
...
+            (CURRENT_DAY - new Date(pr?.pullRequest?.createdAt).getTime()) /
+                ONE_DAY
         ),
-        url: pr.node.url,
-        repoUrl: pr.node.repository.url,
-        title: pr.node.title,
-        createdAt: pr.node.createdAt,
-        avatarUrl: pr.node.author.avatarUrl,
-        project: pr.node.repository.owner
+        url: pr?.pullRequest?.url,
```
Comment (Extheoisah) [inline src/helpers/get-pull-requests.ts:22]: return null or empty string than undefined using ternary???

File: src/hooks/useGithubIssues.ts [hunk 16]
```diff
@@ -45,6 +46,9 @@ export const useGithubIssues = () => {
         mergedPrs: []
     })
 
+    const startDate = new Date(Number(yearlyFilter), 0, 1, 4).toISOString()
```
Comment (Extheoisah) [inline src/hooks/useGithubIssues.ts:49]: what are these values `0, 1, 4`? pls add a comment above

File: src/hooks/useGithubIssues.ts [hunk 17]
```diff
@@ -69,40 +73,65 @@ export const useGithubIssues = () => {
...
+                ranged_issues?.endCursorObj?.end_cursor
+            )
+
             setLoading(false)
 
-            if (issues.error || prs.error) {
-                console.error(issues.error, "error")
-                setError(issues.error[0].message || prs.error[0].message)
+            if (ranged_issues.error || ranged_prs.error) {
+                console.error(ranged_issues.error, "error")
```
Comment (Extheoisah) [inline src/hooks/useGithubIssues.ts:99]:
```suggestion
```

**Review by Extheoisah — CHANGES_REQUESTED** [review]

File: src/graphql/queries.ts (hunk 1 above)
Comment (Emmanuel-Develops) [inline src/graphql/queries.ts:144]: spelling check

```suggestion
export const FETCH_CONTRIBUTION_YEARS = `query contributions($username: String!, ) {
```

**Comment by Extheoisah** [comment]
Bug while testing: contribution years
image

**Comment by IgboPharaoh** [comment]
pushed a fix for the overflowing side panel and corrections

**Review by Extheoisah — APPROVED** [review]

**Pull request merged**